### ISSUE TITLE
Count(?s) query to Stardog is rewritten thus index is not used

### ISSUE THREAD

**Issue by SvenLieber** [comunica/comunica#860]
#### Issue type:

- :bug: Bug 

____
#### Description:

A `count(?subject)` query which is resolved in milliseconds by the native Stardog query interface (probably via an index) takes more than 40 seconds via comunica: it seems comunica rewrites the query and selects all subjects, predicates, and objects.

The query `SELECT (COUNT(?s) AS ?count) WHERE { ?s ?p ?o .}` issued via comunica is shown in the stardog user interface as `SELECT ?s ?p ?o WHERE { ?s ?p ?o.}`

According to the comunica debug log the stardog endpoint (URL ending in `{db name}/query`) *is* recognized as sparql source (thus probably not related to https://github.com/comunica/comunica/issues/535). `Curl`ing also shows me lots of sparql service description triples.

```
[2021-09-24T12:02:58.542Z]  INFO: Identified as sparql source: http://myURL/myDB/query {
  actor: 'https://linkedsoftwaredependencies.org/bundles/npm/@comunica/actor-init-sparql/^1.0.0/config/sets/resolve-sparql.json#mySparqlQuadPatternResolver'
}
```

Forcing sparql as source type results in an `401 - Unauthorized` error even though username and password are provided as described in the [comunica documentation](https://comunica.dev/docs/query/advanced/basic_auth/) (the authentication works when not forcing sparql).

Unfortunately I do not have a public endpoint available, but could provide access on demand for testing.
____
#### Environment:

`comunica-sparql -v` reports the following:

```
| software            | version
| ------------------- | -------
| Comunica Init Actor | 1.22.1 
| node                | v14.2.0
| npm                 | 6.14.4
| yarn                | 1.9.4
| Operating System    | linux (Linux 4.4.0-210-generic)
```

The stardog server has version `7.7.2`

**Comment by rubensworks** [comment]
So there are 2 problems here:

1. The `sparql` source type is not detected because stardog endpoint URLs end with `/query`, so we need to add that suffix in these places:
  * https://github.com/comunica/comunica/blob/master/packages/actor-query-operation-sparql-endpoint/lib/ActorQueryOperationSparqlEndpoint.ts#L65-L66
  * https://github.com/comunica/comunica/blob/master/packages/actor-rdf-resolve-hypermedia-sparql/lib/ActorRdfResolveHypermediaSparql.ts#L25
  * https://github.com/comunica/comunica/blob/master/packages/actor-rdf-update-hypermedia-sparql/lib/ActorRdfUpdateHypermediaSparql.ts#L23-L24
2. Auth headers are not delegated to SPARQL endpoints => we need to pass the proper headers to the endpoint fetcher: https://github.com/comunica/comunica/blob/master/packages/actor-query-operation-sparql-endpoint/lib/ActorQueryOperationSparqlEndpoint.ts#L106-L124

**Comment by rubensworks** [comment]
Should be fixed thanks to #896.
